### PR TITLE
feat(difference, intersection): add multiset variants

### PR DESCRIPTION
**Pull request by eranhirsch**
Adding a variant difference for multiset semantics and update the deprecation warning.

### CODE REVIEW

File: src/difference.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import { purry } from "./purry";
 /**
  * Excludes the values from `other` array.
  *
- * ! **DEPRECATED**: Use `R.filter(array, R.isNot(R.isIncludedIn(other)))`. Will be removed in v2!
+ * ! **DEPRECATED**: The runtime implementation of this function handles duplicate values inconsistently. In v2 a new implementation would replace it; this implementation is accessible to v1 users via the `multiset` variant of this function (`R.difference.multiset`), or to maintain the same runtime implementation use `R.filter(array, R.isNot(R.isIncludedIn(other)))` instead.
```
Comment (cjquines) [inline src/difference.ts:8]: **suggestion:**

```suggestion
 * ! **DEPRECATED**: In v2, `R.difference.multiset` will replace `R.difference`. To keep current behavior, use `R.filter(array, R.isNot(R.isIncludedIn(other)))` instead.
```

**Review by cjquines — COMMENTED** [review]
let's include an example with duplicate values to illustrate the difference?

**Review by cjquines — APPROVED** [review]

**Comment by cjquines** [comment]
hooray 🎉 

does this make #368 no longer needed? are multiset semantics going to be our default moving-forward, and does that mean we want the other multiset helpers, or what?

**Comment by eranhirsch** [comment]
> does this make https://github.com/remeda/remeda/pull/368 no longer needed?

Other utilities we don't offer or don't offer correctly would need to be ported, too. For now, it's a good template for me to copy from and a reminder to do it.

**Pull request merged**